plugins/modules/ndo_fabric_span_session.py

```diff
@@ -0,0 +1,698 @@
+                    mso_update_values[("destination", "local")] = dict()
+
+                mso_update_values[("destination", "local")] = dict(accessInterface=destination_port_uuid)
+
+            if destination_port_channel:
+                if match.details.get("destination", {}).get("remote", {}).get("epgRef"):
+                    mso_update_values[("destination", "remote")] = dict()
+
+                if match.details.get("destination", {}).get("local", {}).get("accessInterface"):
+                    mso_update_values[("destination", "local")] = dict()
+
+                mso_update_values[("destination", "local")] = dict(portChannel=destination_port_channel_uuid)
+
+            proposed_payload = copy.deepcopy(match.details)
+            append_update_ops_data(ops, proposed_payload, span_session_path, mso_update_values)
+            mso.sanitize(proposed_payload, collate=True)
+        else:
+            mso_values["name"] = name
+            mso_values["description"] = description
+
+            if admin_state is not None:
+                mso_values["sourceGroup"] = dict(enableAdminState=admin_state)
+
+            mso.sanitize(mso_values)
+            ops.append(dict(op="add", path=span_session_path, value=mso.sent))
+
+    elif state == "absent":
+        if match:
+            ops.append(dict(op="remove", path=span_session_path))
+
+    if not module.check_mode and ops:
+        response = mso.request(mso_template.template_path, method="PATCH", data=ops)
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            response.get("monitoringTemplate", {}).get("template", {}).get("spanSessions", []),
+            [KVPair("uuid", uuid) if uuid else KVPair("name", name)],
+        )
+        if match:
+            set_fabric_span_session_object_details(mso_template, site_id, match.details)
+            mso.existing = match.details  # When the state is present
+        else:
+            mso.existing = {}  # When the state is absent
+    elif module.check_mode and state != "query":  # When the state is present/absent with check mode
+        set_fabric_span_session_object_details(mso_template, site_id, mso.proposed)
+        mso.existing = mso.proposed if state == "present" else {}
+    mso.exit_json()
+
+
+def set_fabric_span_session_object_details(mso_template, site_id, span_session):
+    span_session.update({"templateId": mso_template.template_id, "templateName": mso_template.template_name})
+    if span_session.get("destination", {}).get("local", {}).get("accessInterface"):
+        interface = get_site_interface_details(mso_template.mso, site_id, span_session.get("destination").get("local").get("accessInterface"))
+        interface.pop("uuid", None)
+        span_session.get("destination").get("local").update(interface)
+    elif span_session.get("destination", {}).get("local", {}).get("portChannel"):
```

Comment:
You may not need these two `elif` statements now that the `update_config_with_template_and_references` function recursively finds matching reference properties.

Reply:
Done!